### ISSUE TITLE
Search random hex

### ISSUE THREAD

**Issue by dtoan1402** [JeanLucPons/VanitySearch#20]
May i ask if VanitySearch work by generate massive sha256 random hex priv key? If there isn't, i'd like to modify it

**Comment by broka77** [comment]
+1

**Comment by ttran2425** [comment]
> May i ask if VanitySearch work by generate massive sha256 random hex priv key? If there isn't, i'd like to modify it

I saw many ppl demand this feature on many brute force & vanitygen app. What are your intention with this?

**Comment by JeanLucPons** [comment]
Hi there,
This would require a different program, at least for the GPU side.
This feature requires to compute a complete EC multiplication at each point.
The GPU constant memory is small. The current kernel performs only addition and use the constant memory for storing consecutive generators. A GPU kernel kernel designed for EC multiplication should use the constant memory for storing a multiplicative generators table.
If you want to modify this, It would be better to start a new project.

**Comment by dtoan1402** [comment]
> Hi there,
> This would require a different program, at least for the GPU side.
> This feature requires to compute a complete EC multiplication at each point.
> The GPU constant memory is small. The current kernel performs only addition and use the constant memory for storing consecutive generators. A GPU kernel kernel designed for EC multiplication should use the constant memory for storing a multiplicative generators table.
> If you want to modify this, It would be better to start a new project.

Well, i tried to write one myself run on CPU with C# & it can only run 60k keys/sec which is why i'm asking if i can modify your app since your si new. Most of bitcoin brute force or key generator i know only generate one single private key then mutiply the public keys which practically never reach any collisions if you ever generate 2^80 keys

**Comment by JeanLucPons** [comment]
Yes you can modify it but as I said It will better to start a new program, the strategy will be very different than the one currently used in VanitySearch but don't expect a very high key rate. EC multiplication is slow.

**Comment by broka77** [comment]
@JeanLucPons Can you start such a project? by pollard_rho method or something else for brute force BTC ... please

**Comment by broka77** [comment]
Please create a new project for brute-force Bitcoin addresses

**Comment by JeanLucPons** [comment]
When i have time, i'll start this pollard_rho project.

**Comment by broka77** [comment]
@JeanLucPons  Please everyone is waiting for this :) please take some time off

**Comment by neutron220** [comment]
@broka77 don't be so annoying like a little kid, with please word. 
I suggest you to ask JeanLucPons for BTC donation address, and donate him for his hard work and for answers than he gave to you. (at most he was answering your's questions).
Because you said, you have time and money, so why didn't share ?

**Comment by broka77** [comment]
@JeanLucPons 
Any news on the Pollard rho method? Are you planning to develop a brute forcer?

**Comment by JeanLucPons** [comment]
Very busy at the moment, huge professional work, very difficult for me to find free time.

**Comment by broka77** [comment]
@JeanLucPons  can i write you an email? do you have e-Mail?

**Comment by alevlaslo** [comment]
> @JeanLucPons Сможете ли вы начать такой проект? методом pollard_rho или чем-то еще для брутфорса BTC ... пожалуйста

https://github.com/traxm/Bitcoin-Micro-Collider
